### PR TITLE
Issue 2091 - D2 final cannot be applied to variable

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=2091

To lift the limitation, display the error only when `final` is _directly_ applied on a variable.
Use same method to `synchrionized`, `abstract`, and `override`.

### CODE REVIEW

File: src/declaration.c [hunk 1]
```diff
@@ -1107,24 +1109,17 @@ void VarDeclaration::semantic(Scope *sc)
...
-    }
-    else if (isAbstract())
-    {
-        error("abstract cannot be applied to variable");
-    }
-    else if (storage_class & STCfinal)
-    {
-        error("final cannot be applied to variable, perhaps you meant const?");
+        if (stc == STCfinal)
+            error("cannot be final, perhaps you meant const?");
```
Comment (Hackerpilot) [inline src/declaration.c]: Will the message output to the user contain the name of the variable?

**Comment by Hackerpilot** [comment]
This seems reasonable to me. A change should be made to the "Attributes" page of the language specification if this pull request is merged to mention that `final`, `synchronized`, `abstract` and `override` do not apply to variable declarations.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks!

**Pull request merged**